redash/tasks/queries.py

```diff
@@ -479,7 +479,7 @@ def run(self):
 
         _unlock(self.query_hash, self.data_source.id)
 
-        if error:
+        if error is not None:
```

Comment:
```suggestion
        if error is not None and data is None:
```

Maybe also make sure `data` is `None`?

Reply:
done